src/libsyntax/ext/build.rs

```diff
@@ -567,19 +563,18 @@ impl<'a> AstBuilder for ExtCtxt<'a> {
     }
 
     fn block_expr(&self, expr: P<ast::Expr>) -> P<ast::Block> {
-        self.block_all(expr.span, Vec::new(), Some(expr))
-    }
-    fn block_all(&self,
-                 span: Span,
-                 stmts: Vec<ast::Stmt>,
-                 expr: Option<P<ast::Expr>>) -> P<ast::Block> {
-            P(ast::Block {
-               stmts: stmts,
-               expr: expr,
-               id: ast::DUMMY_NODE_ID,
-               rules: BlockCheckMode::Default,
-               span: span,
-            })
+        self.block_all(expr.span, vec![Spanned {
```

Comment:
This won't land directly -- once it's approved I'll rebase over #34316 in the breaking batch, so this will become `ast::Stmt { span: expr.span, node: StmtKind::Expr(expr), id: DUMMY_NODE_ID }`.